### ISSUE TITLE
Web hosting through Syncloud? Feature request

### ISSUE THREAD

**Issue by chegeiser** [syncloud/platform#536]
It would be great to be able to harness Syncloud SSL cert processing to be able to set up additional custom webpages/web apps at custom sub domains. For example, if I own myaddress.com and want to host mywebapp.myaddress.com I’d like to use Syncloud to set this up. This could be hosted on the Syncloud file server or on another file server behind my firewall with Syncloud pointing to it. 

Is this a possible new feature for Syncloud?

Thank you!

**Comment by chegeiser** [comment]
Would there be a way to harness Syncloud’s ssl creation for this custom address?

**Comment by cyberb** [comment]
Right, https certificate is actually not wildcard right now so it will not cover random app names.

Wildcard support should arrive after this ticket (https://github.com/syncloud/platform/issues/461).

Another question, do you want to host only static files or alsp dynamic like php?
If only static probably there is some.app wich simplifies static file hpsting which we can package as an app.

**Comment by chegeiser** [comment]
My web app syncs data to CouchDB. Currently this is not hosted on the Syncloud device, but leveraging the wildcard ssl cert would be very useful. I could still keep CouchDB on another device and have the webhook point to that machine.

**Comment by cyberb** [comment]
If certificate is the only problem should we close the request?

**Comment by chegeiser** [comment]
I think so, as long as the outline below is accurate.
1. Currently can edit platform/config/nginx/public.conf to add custom web app subdomain name: `[test123] => /var/snap/[name]/common/web.socket`  and nginx will then point that subdomain to `/var/snap/test123/common/web.socket`
2. Support for ssl wildcard for subdomains will be added per ticket #461.

**Comment by cyberb** [comment]
There is no need to edit anything, name to web socket is already dynamic in the config.

**Comment by chegeiser** [comment]
Got it. When I look at the web.socket files for installed apps (like Gogs, Files, etc) the web.socket file is empty. How do I get thisapp.mydomain.com to direct to an index.html file like `/var/snap/thisapp/common/index.html'

**Comment by cyberb** [comment]
web.socket file is a a special unix domain socket created by a web server (of test app) for serving files.
You will have to start a web server (nginx for example) and listen on this file. Then you can configure this web server to serve your files.

Example of an app nginx config:  https://github.com/syncloud/users/blob/master/config/nginx.conf#L25

**Comment by chegeiser** [comment]
So it can’t be served by the nginx instance running all the Syncloud apps?

**Comment by cyberb** [comment]
Platform nginx is only service its web files under ```/snap/platform/current/www``` but this is read-only file system for security reasons.

You could probably change nginx config to add another location as a test (it will reset every system upgrade)

Ideally we can find some open-source project that simplifies static html hosting and uploading of hosted files using http protocol, package it as an app called www and that will be nice www.[device] domain name with some custom user html.

**Comment by cyberb** [comment]
Or you run nginx with a simple config to serve files from where ever you want and listen on any name you like.

**Comment by chegeiser** [comment]
The first option sounds great as it would provide the simplest method for users posting their own static pages using Syncloud as a tool to do that. The second option could work, but does require the user to really understand what they are doing. (I fall in the category of partially understanding some of what is needed to meet option 2).

**Comment by cyberb** [comment]
One more thing, existing tools like WordPress can actually serve static html files, have you tried that:

https://blog.hubspot.com/website/upload-html-file-to-wordpress

**Comment by simoneburalli** [comment]
> One more thing, existing tools like WordPress can actually serve static html files, have you tried that:
> 
> https://blog.hubspot.com/website/upload-html-file-to-wordpress

Hi Boris,
I have read the guide but Wordpress Admin dashboard cannot be reached to its standard address (https://.../wp-admin/). Could you kindly tell me what the correct address of the dashboard is? Thanks a lot

**Comment by cyberb** [comment]
WordPress login is /wp-login.php
For example: https://wordpress.[device].syncloud.it/wp-login.php

I think they had a button before but not now, somehting must have changed in the latest update.
